pkg/reconciler/revision/resources/deploy.go

```diff
@@ -108,7 +110,7 @@ func rewriteUserProbe(p *corev1.Probe, userPort int) {
 	}
 }
 
-func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, tracingConfig *tracingconfig.Config, observabilityConfig *metrics.ObservabilityConfig, deploymentConfig *deployment.Config) (*corev1.PodSpec, error) {
+func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, tracingConfig *tracingconfig.Config, observabilityConfig *metrics.ObservabilityConfig, deploymentConfig *deployment.Config, secretLister v1.SecretLister) (*corev1.PodSpec, error) {
```

Comment:
Why put this all the way down in makePodSpec? This seems like it might be more appropriate to do after `MakeDeployment` instead of plumbing it down within it. MakeDeployment today is fairly lightweight doing defaulting and transformations from a Revision to a Deployment. Validation of the Deployment seems like an independent step to do between Make and Create (or Update).